goinstant/docker.go

```diff
@@ -187,3 +194,182 @@ func RunDockerCommand(commandSlice ...string) {
+			return
+		}
+
+		if zipRegex.MatchString(pathToPackage) {
+			pathToPackage = unzipPackage(resp.Body)
+		} else if tarRegex.MatchString(pathToPackage) {
+			pathToPackage = untarPackage(resp.Body)
+		}
+	}
+
+	commandSlice := []string{"cp", pathToPackage, "instant-openhie:instant/"}
+	RunDockerCommand(commandSlice...)
+}
+
+func createZipFile(file string, content io.Reader) {
+	output, err := os.Create(file)
+	defer output.Close()
+
+	_, err = io.Copy(output, content)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error in creating zip file:", err)
+		return
+	}
+}
+
+func unzipPackage(zipContent io.ReadCloser) (pathToPackage string) {
+	tempZipFile := "temp.zip"
+	createZipFile(tempZipFile, zipContent)
+
+	// Unzip file
+	archive, err := zip.OpenReader(tempZipFile)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error in unzipping file:", err)
+		return
+	}
+
+	packageName := ""
+	for _, file := range archive.File {
+		filePath := filepath.Join(".", file.Name)
+
+		if file.FileInfo().IsDir() {
+			if packageName == "" {
+				packageName = file.Name
+			}
+			os.MkdirAll(filePath, os.ModePerm)
+			continue
+		}
+
+		content, err := file.Open()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Error in unzipping file:", err)
+			return
+		}
+
+		dest, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Error in unzipping file:", err)
+			return
+		}
+		_, err = io.Copy(dest, content)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Error in copying unzipped files:", err)
+			return
+		}
+		content.Close()
+	}
+
+	// Remove temp zip file
+	tempFilePath := filepath.Join(".", tempZipFile)
+	archive.Close()
+	err = os.Remove(tempFilePath)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error in deleting temp.zip file:", err)
+		return
+	}
+
+	pathToPackage = filepath.Join(".", packageName)
+	return
+}
+
+func untarPackage(tarContent io.ReadCloser) (pathToPackage string) {
+	packageName := ""
+	gzipReader, err := gzip.NewReader(tarContent)
+	if err != nil {
+		return
+	}
+	defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+
+	for {
+		file, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if file == nil {
+			continue
+		}
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Error in extracting tar file:", err)
+			continue
+		}
+
+		filePath := filepath.Join(".", file.Name)
+		if file.Typeflag == tar.TypeDir {
+			if packageName == "" {
+				packageName = filePath
+			}
+			os.MkdirAll(filePath, os.ModePerm)
+			continue
+		}
+
+		dest, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0755)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Error in untaring file:", err)
+			return
+		}
+		if _, err := io.Copy(dest, tarReader); err != nil {
+			fmt.Fprintln(os.Stderr, "Error in extracting tar file:", err)
+		}
+	}
+	pathToPackage = filepath.Join(".", packageName)
+	return
+}
+
+func retrieveGitRepo(gitUrl string) (pathToPackage string) {
```

Comment:
TODO: I will refactor this to use the same logic as the docker command, no need to duplicate this logic.

Reply:
Done